app/models/claim/claims_preventing_payment_finder.rb

```diff
@@ -0,0 +1,29 @@
+class Claim
+  class ClaimsPreventingPaymentFinder
+    attr_reader :claim
+
+    def initialize(claim)
+      @claim = claim
+    end
+
+    def claims_preventing_payment
+      @claims_preventing_payment ||= find_claims_preventing_payment
+    end
+
+    def find_claims_preventing_payment
+      payrollable_claims_from_same_claimant = Claim.payrollable.where(national_insurance_number: claim.national_insurance_number)
+
+      payrollable_claims_from_same_claimant.select do |other_claim|
+        Payment::PERSONAL_DETAILS_ATTRIBUTES_FORBIDDING_DISCREPANCIES.any? do |attribute|
+          other_claim.read_attribute(attribute) != claim.read_attribute(attribute)
```

Comment:
Given that this logic is used is two places, is it worth breaking this into a method? Something like:

```ruby
def attribute_does_not_match?(claim_to_compare, attribute)
  claim_to_compare.read_attribute(attribute) != claim.read_attribute(attribute)
end
```

Reply:
Sure thing, will change.